src/ArrayWalker.php

```diff
@@ -20,11 +20,12 @@ class ArrayWalker
      *
      * @return mixed The matching element's value if found, otherwise null.
      */
-    public static function walk(array $array, array $path)
+    public static function &walk(array &$array, array $path)
     {
         while (count($path)) {
             if (!is_array($array)) {
-                return null;
+                $array = null;
+                break;
```

Comment:
Why is this change needed? Does this modify the input array? If so, why? If not, what's the difference between just returning `null`?